kubernetes/validators.go

```diff
@@ -11,6 +11,18 @@ import (
 	utilValidation "k8s.io/apimachinery/pkg/util/validation"
 )
 
+func addAnnotationToWhiteList(value interface{}, key string) (ws []string, es []error) {
+	errors := utilValidation.IsQualifiedName(strings.ToLower(value.(string)))
+	if len(errors) > 0 {
+		for _, e := range errors {
+			es = append(es, fmt.Errorf("invalid annotation key %s (%q) %s", key, value.(string), e))
+		}
+	} else {
+		permittedInternalAnnotations[value.(string)] = true
```

Comment:
Validators should not mutate anything.
Can you explain a bit why are you using the validator function to populate the list?
I see you're already doing that in the beginning of providerConfigure().

Reply:
Validation runs before providerConfigure. If the validation fails, the provider won't configured... I know, it a kind of hack, but I didn't find better place to hook it.